source/store/new_graph_store.py

```diff
@@ -13,227 +17,388 @@ def __init__(self):
+            return
+        if dialog.dict['d'].text().isnumeric() and dialog.dict['n'].text().isnumeric() and \
+                (int(dialog.dict['d'].text()) * int(dialog.dict['n'].text())) % 2 == 0:
+            dialog.create_button.setEnabled(True)
+            dialog.dict['d'].setStyleSheet('background-color: white;')
+            dialog.dict['n'].setStyleSheet('background-color: white;')
+        else:
+            dialog.create_button.setDisabled(True)
+            dialog.dict['d'].setStyleSheet('background-color: #EF5350;')
+            dialog.dict['n'].setStyleSheet('background-color: #EF5350;')
 
 
 class RandomCograph(NewGraphStore):
     name = "Random Cograph"
+    dict_attributes_names = {"n": "K value for a graph with 2^k nodes (k)", "conditions": "0 < k <= 60"}
 
     @staticmethod
     def open_dialog():
-        dialog = NewGraphDialog(name=RandomCograph.name, n='')
-        dialog.dialog_next_button.clicked.connect(lambda: RandomCograph.create_graph(dialog))
-        dialog.exec()
+        dialog = NewGraphDialog(RandomCograph.dict_attributes_names, name=RandomCograph.name, n='')
+        dialog.create_button.clicked.connect(lambda: RandomCograph.create_graph(dialog))
+        dialog.graph_link.clicked.connect(lambda: NewGraphStore.open_url
+                                          ("https://en.wikipedia.org/wiki/Cograph"))
+        dialog.dict['n'].textEdited.connect(lambda: NewGraphStore.verify_natural_number(dialog, 'n'))
+
+        NewGraphStore.open_new_dialog(dialog)
 
     @staticmethod
     def create_graph(dialog):
         new_graph_store.set_graph(nx.random_cograph(int(dialog.dict['n'].text())))
-        new_graph_store.set_file_path(project_information_store.project_location + f"\\{dialog.dict['name'].text()}.g6")
+        NewGraphStore.create_graph(dialog)
 
-        dialog.close()
+
+class CompleteBipartiteGraph(NewGraphStore):
+    name = "Complete Bipartite Graph"
+    dict_attributes_names = {"m": "Number of nodes for set A (m)", "n": "Number of nodes for set B (n)",
+                             "conditions": "0 < m,n <= 60"}
+
+    @staticmethod
+    def open_dialog():
+        dialog = NewGraphDialog(CompleteBipartiteGraph.dict_attributes_names,
+                                name=CompleteBipartiteGraph.name, m='', n='')
+        dialog.create_button.clicked.connect(lambda: CompleteBipartiteGraph.create_graph(dialog))
+        dialog.graph_link.clicked.connect(lambda: NewGraphStore.open_url
+                                          ("https://en.wikipedia.org/wiki/Complete_bipartite_graph"))
+        dialog.dict['m'].textEdited.connect(lambda: NewGraphStore.verify_natural_number(dialog, 'm'))
+        dialog.dict['n'].textEdited.connect(lambda: NewGraphStore.verify_natural_number(dialog, 'n'))
+
+        NewGraphStore.open_new_dialog(dialog)
+
+    @staticmethod
+    def create_graph(dialog):
+        new_graph_store.set_graph(nx.complete_bipartite_graph(int(dialog.dict['m'].text()),
```

Comment:
Change to [Multipartite Complete](https://networkx.org/documentation/stable/reference/generated/networkx.generators.classic.complete_multipartite_graph.html#networkx.generators.classic.complete_multipartite_graph), with 2 sets.

Reply:
Done!